tests/integration_test.rs

```diff
@@ -1702,7 +1691,105 @@ fn test_near_call_callee_uses_gas() {
 fn test_near_call_callee_less_gas() {
     let bin_path = make_bin_path_asm("near_call_callee_less_gas");
     let (result, _) = run_program(&bin_path);
-    assert_eq!(result, U256::from(1));
+    assert_eq!(result, U256::from(6));
+}
+
+#[test]
+fn test_near_call_revert() {
+    let bin_path = make_bin_path_asm("near_call_revert");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(6));
+}
+
+#[test]
+fn test_near_call_revert_stack() {
+    let bin_path = make_bin_path_asm("near_call_revert_stack");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+fn test_near_call_revert_heap() {
+    let bin_path = make_bin_path_asm("near_call_revert_heap");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+fn test_near_call_panic_heap() {
+    let bin_path = make_bin_path_asm("near_call_panic_heap");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+fn test_near_call_revert_aux_heap() {
+    let bin_path = make_bin_path_asm("near_call_revert_aux_heap");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+fn test_near_call_panic_aux_heap() {
+    let bin_path = make_bin_path_asm("near_call_panic_aux_heap");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+#[should_panic = "Contract Reverted"]
+fn test_revert() {
+    let bin_path = make_bin_path_asm("revert");
+    run_program(&bin_path);
+}
+
+#[test]
+fn test_near_call_panic() {
+    let bin_path = make_bin_path_asm("near_call_panic");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(6));
+}
+
+#[test]
+fn test_near_call_panic_stack() {
+    let bin_path = make_bin_path_asm("near_call_panic_stack");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(5));
+}
+
+#[test]
+#[should_panic = "Contract Panicked"]
+fn test_panic() {
+    let bin_path = make_bin_path_asm("panic");
+    run_program(&bin_path);
+}
+
+#[test]
+fn test_near_call_panic_spends_gas() {
+    let bin_path = make_bin_path_asm("near_call_panic_spends_gas");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(6));
+}
+
+#[test]
+fn test_near_call_returns_with_label() {
+    let bin_path = make_bin_path_asm("near_call_returns_with_label");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(6));
+}
+
+#[test]
+fn test_near_call_reverts_with_label() {
+    let bin_path = make_bin_path_asm("near_call_revert_with_label");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(7));
+}
+
+#[test]
+fn test_near_call_panics_with_label() {
+    let bin_path = make_bin_path_asm("near_call_panics_with_label");
+    let (result, _) = run_program(&bin_path);
+    assert_eq!(result, U256::from(7));
```

Comment:
Always nice to see something extended extensively, great work! 🚀 .

Reply:
Thanks!